### PR TITLE
Erc20 forwarder - idle finance feature add

### PR DESCRIPTION
**Pull request by livingrock7**
permit chained execution

check allowance only if permti chained is not requested.
added method for permit chained execution that allows metaInfo to be passed
cost calculation component for permit cost 

further changes subject to review and testing

### CODE REVIEW

File: src/ERC20ForwarderClient.js [hunk 1]
```diff
@@ -496,6 +556,102 @@ class ERC20ForwarderClient {
...
+        signature == null
+          ? await this.provider.send("eth_signTypedData_v3", [
+              req.from,
+              JSON.stringify(dataToSign),
+            ])
+          : signature;
+      const api = this.getApiId(req);
+      if (!api || !api.id)
+        throw new Error(
+          "Could not find the apiId for the given request. Contact Biconomy for resolution"
```
Comment (tomarsachin2271) [inline src/ERC20ForwarderClient.js:623]: Change message to "Could not find the method information on Biconomy Dashboard. Check if you have registered your method on the Dashboard."

File: src/ERC20ForwarderClient.js [hunk 2]
```diff
@@ -509,6 +665,28 @@ class ERC20ForwarderClient {
...
   async sendTxPersonalSign({ req, signature = null, userAddress }) {
     try {
+      // should we?
+      // check allowance at each final call
+      // needs spendValue that is cost
+
+      /*if(!userAddress)
+        userAddress = await this.provider.getSigner().getAddress();
+
+      const allowedToSpend = await this.erc20ForwarderApproved(
```
Comment (tomarsachin2271) [inline src/ERC20ForwarderClient.js:675]: Why this was commented?

File: src/ERC20ForwarderClient.js (hunk 1 above)
Comment (livingrock7) [reply src/ERC20ForwarderClient.js:623]: ok done.

File: src/ERC20ForwarderClient.js (hunk 2 above)
Comment (livingrock7) [reply src/ERC20ForwarderClient.js:675]: it wasnt there in sendTx methods
cause, allowedToSpend is checked in buildTx


now there are two ways (the problem/reason is buildTx shouldnt check allowed to spend and reject if it is meant for permit chained execution) 

i) move allowance check to sendTx stages from buildTx
ii) in buildTx check based on permitType (whether its passed or not) then only check allowance  

ii is currently done now. Also increased cost is displayed if permitType is passed

Comment (livingrock7) [reply src/ERC20ForwarderClient.js:675]: any thoughts? check buildTx changes also in a file

**Review by tomarsachin2271 — APPROVED** [review]

**Review by tomarsachin2271 — APPROVED** [review]

**Pull request merged**